code/game/objects/items/robot/items/hypo.dm

```diff
@@ -175,32 +175,35 @@
 					cyborg.cell.use(charge_cost)
 					stored_reagents.add_reagent(reagent_to_regen, 5, reagtemp = dispensed_temperature, no_react = TRUE)
 
-/obj/item/reagent_containers/borghypo/attack(mob/living/carbon/injectee, mob/user)
-	if(!istype(injectee))
-		return
+/obj/item/reagent_containers/borghypo/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if(!iscarbon(interacting_with))
+		return NONE
+	var/mob/living/carbon/injectee = interacting_with
 	if(!selected_reagent)
 		balloon_alert(user, "no reagent selected!")
-		return
+		return ITEM_INTERACT_BLOCKING
 	if(!stored_reagents.has_reagent(selected_reagent.type, amount_per_transfer_from_this))
 		balloon_alert(user, "not enough [selected_reagent.name]!")
-		return
+		return ITEM_INTERACT_BLOCKING
 
-	if(injectee.try_inject(user, user.zone_selected, injection_flags = INJECT_TRY_SHOW_ERROR_MESSAGE | (bypass_protection ? INJECT_CHECK_PENETRATE_THICK : 0)))
-		// This is the in-between where we're storing the reagent we're going to inject the injectee with
-		// because we cannot specify a singular reagent to transfer in trans_to
-		var/datum/reagents/hypospray_injector = new()
-		stored_reagents.remove_reagent(selected_reagent.type, amount_per_transfer_from_this)
-		hypospray_injector.add_reagent(selected_reagent.type, amount_per_transfer_from_this, reagtemp = dispensed_temperature, no_react = TRUE)
-
-		to_chat(injectee, span_warning("You feel a tiny prick!"))
-		to_chat(user, span_notice("You inject [injectee] with the injector ([selected_reagent.name])."))
-
-		if(injectee.reagents)
-			hypospray_injector.trans_to(injectee, amount_per_transfer_from_this, transferred_by = user, methods = INJECT)
-			balloon_alert(user, "[amount_per_transfer_from_this] unit\s injected")
-			log_combat(user, injectee, "injected", src, "(CHEMICALS: [selected_reagent])")
-	else
+	if(!injectee.try_inject(user, user.zone_selected, injection_flags = INJECT_TRY_SHOW_ERROR_MESSAGE | (bypass_protection ? INJECT_CHECK_PENETRATE_THICK : 0)))
 		balloon_alert(user, "[injectee.parse_zone_with_bodypart(user.zone_selected)] is blocked!")
+		return ITEM_INTERACT_BLOCKING
+
+	// This is the in-between where we're storing the reagent we're going to inject the injectee with
+	// because we cannot specify a singular reagent to transfer in trans_to
+	var/datum/reagents/hypospray_injector = new()
+	stored_reagents.remove_reagent(selected_reagent.type, amount_per_transfer_from_this)
+	hypospray_injector.add_reagent(selected_reagent.type, amount_per_transfer_from_this, reagtemp = dispensed_temperature, no_react = TRUE)
```

Comment:
We can, see ``target_id``, although you'll need to fix ``transfer_amount = reagent.volume * part`` in trans_to code ignoring actual desired amount

Reply:
I wanted to avoid refactoring this unnecessarily because I didn't want to test borgs